engines/wintermute/base/gfx/x/frame_node.cpp

```diff
@@ -0,0 +1,459 @@
+		_lerpValue[i] = 0.0f;
+
+		_transUsed[i] = false;
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////
+FrameNode::~FrameNode(void) {
+	// remove child frames
+	for (uint32 i = 0; i < _frames.size(); i++) {
+		delete _frames[i];
+	}
+
+	_frames.clear();
+
+	// remove meshes
+	for (uint32 i = 0; i < _meshes.size(); i++) {
+		delete _meshes[i];
+	}
+
+	_meshes.clear();
+}
+
+//////////////////////////////////////////////////////////////////////////
+Math::Matrix4 *FrameNode::getCombinedMatrix() {
+	return &_combinedMatrix;
+}
+
+//////////////////////////////////////////////////////////////////////////
+Math::Matrix4 *FrameNode::getOriginalMatrix() {
+	return &_originalMatrix;
+}
+
+//////////////////////////////////////////////////////////////////////////
+void FrameNode::setTransformationMatrix(Math::Matrix4 *mat) {
+	_transformationMatrix = *mat;
+}
+
+//////////////////////////////////////////////////////////////////////////
+void FrameNode::setTransformation(int slot, Math::Vector3d pos, Math::Vector3d scale, Math::Quaternion rot, float lerpValue) {
+	if (slot < 0 || slot > 1)
+		return;
+
+	_transPos[slot] = pos;
+	_transScale[slot] = scale;
+	_transRot[slot] = rot;
+	_lerpValue[slot] = lerpValue;
+
+	_transUsed[slot] = true;
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool FrameNode::loadFromX(const Common::String &filename, XFileLexer &lexer, ModelX *model) {
+	_gameRef->miniUpdate();
+
+	bool ret = true;
+
+	setName(lexer.tokenToString().c_str());
+	lexer.advanceToNextToken();
+	lexer.advanceOnOpenBraces();
+
+	while (!lexer.eof()) {
+		if (lexer.tokenIsIdentifier("Frame")) {
+			lexer.advanceToNextToken();
+			FrameNode *child = new FrameNode(_gameRef);
+			if (child->loadFromX(filename, lexer, model)) {
+				_frames.add(child);
+			} else {
+				delete child;
+			}
+		} else if (lexer.tokenIsIdentifier("Mesh")) {
+			lexer.advanceToNextToken();
+			MeshX *mesh = new MeshX(_gameRef);
+
+			if (mesh->loadFromX(filename, lexer)) {
+				_meshes.add(mesh);
+			} else {
+				delete mesh;
+			}
+		} else if (lexer.tokenIsIdentifier("FrameTransformMatrix")) {
+			lexer.advanceToNextToken();
+			lexer.advanceToNextToken(); // skip optional name
+			lexer.advanceOnOpenBraces();
+
+			// TODO: check if this is the right format
+			for (int r = 0; r < 4; ++r) {
+				for (int c = 0; c < 4; ++c) {
+					_transformationMatrix(c, r) = readFloat(lexer);
+				}
+			}
+
+			lexer.advanceToNextToken();
+			lexer.advanceToNextToken();
+
+		} else if (lexer.reachedClosedBraces()) {
+			lexer.advanceToNextToken();
+			break;
+		} else {
+			warning("FrameNode::loadFromX unexpected %i token excountered", lexer.getTypeOfToken());
+			ret = false;
+			break;
+		}
+	}
+
+	// {9E415A43-7BA6-4a73-8743-B73D47E88476}
+	//	GUID DXFILEOBJ_AnimTicksPerSecond = { 0x9e415a43, 0x7ba6, 0x4a73, { 0x87, 0x43, 0xb7, 0x3d, 0x47, 0xe8, 0x84, 0x76 } };

```

Comment:
What's this?

Reply:
Left over from wme, but not used anymore. Will remove it.